### ISSUE TITLE
Flaky test

### ISSUE THREAD

**Issue by nazar-pc** [sifraitech/rust-kzg#193]
I had a CI run in my fork that failed, while others (including restart) succeeded with blst_from_scratch: https://github.com/subspace/rust-kzg/actions/runs/4303250577/jobs/7502752040

```
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :clean UP-TO-DATE
> Task :compileJava
> Task :processResources
> Task :classes
> Task :jar
> Task :compileTestFixturesJava
> Task :processTestFixturesResources
> Task :testFixturesClasses
> Task :testFixturesJar
> Task :compileTestJava
> Task :processTestResources NO-SOURCE
> Task :testClasses
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "blst_p1_uncompress failed"', blst-from-scratch/src/eip_4844.rs:60:40

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> Task :test
fatal runtime error: failed to initiate panic, error 5
```

Here is the unwrap that failed (link to main branch, but failed on upcoming `update_upstream` branch):
https://github.com/sifraitech/rust-kzg/blob/fe9cc21a4245cdfbff6de45998570380c41e5257/blst-from-scratch/src/eip_4844.rs#L54-L59

**Comment by sauliusgrigaitis** [comment]
Interesting, could you try to run the `main` branch multiple times? We didn't notice flakiness before.

**Comment by nazar-pc** [comment]
I don't even have enough dependencies installed locally to test all of the tests, just noticed this in CI once and decided to report back.

**Comment by sauliusgrigaitis** [comment]
You could trigger this multiple times on GitHub CI and see how often it happens. It's strange that we didn't experience before.

**Comment by nazar-pc** [comment]
I think I'll need to create a branch that runs just that single Java test over and over again, the whole pipeline takes about an hour and I believe GitHub Actions will kill the job that runs over 4 or 6 hours, can't recall off top of my head.

**Comment by nazar-pc** [comment]
After multiple attempts another CI run failed in exactly the same way, this time on C bindings: https://github.com/subspace/rust-kzg/actions/runs/4322232985/jobs/7544394170

I'm not planning to use eip-4844 module, so it is somewhat low priority for me. Most of the time it succeeds, but after enough attempts it does fail eventually.

**Comment by sauliusgrigaitis** [comment]
Interesting. Let's have a look at it after #191 is resolved.

**Comment by belijzajac** [comment]
Will be fixed by d7b6982967d86745dcf893acba05c92db27dc487 once #207 is merged.
